src/Native/Unix/System.Security.Cryptography.Native/pal_bio.cpp

```diff
@@ -52,3 +52,98 @@ extern "C" int32_t CryptoNative_BioCtrlPending(BIO* bio)
     assert(result <= INT32_MAX);
     return static_cast<int32_t>(result);
 }
+
+extern "C" void CryptoNative_BioSetAppData(BIO* bio, void* data)
+{
+    BIO_set_ex_data(bio, 0, data);
+}
+
+extern "C" void CryptoNative_BioSetWriteFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_WRITE);
+}
+
+extern "C" void CryptoNative_BioSetShoudRetryReadFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_SHOULD_RETRY | BIO_FLAGS_READ);
+}
+
+typedef struct ReadWriteMethodStruct
+{
+    BioWriteCallback write;
+    BioReadCallback read;
+} ReadWriteMethodStruct;
+
+static ReadWriteMethodStruct managedMethods = {
+    nullptr,
+    nullptr
+};
+
+static long ControlCallback(BIO* bio, int cmd, long param, void* ptr)
+{
+    (void)bio, (void)param, (void)ptr; // deliberately unused parameters
+    switch (cmd)
+    {
+        case BIO_CTRL_FLUSH:
+        case BIO_CTRL_POP:
+        case BIO_CTRL_PUSH:
+            return 1;
+    }
+    return 0;
+}
+
+static int DestroyCallback(BIO* bio)
+{
+    (void)bio; // deliberately unused parameter
+    return -1;
+}
+
+static int CreateCallback(BIO* bio)
+{
+    bio->init = 1;
+    return 1;
+}
+
+static int WriteCallback(BIO* b, const char* buf, int32_t len)
+{
+    void* ptr = BIO_get_ex_data(b, 0);
+    if(ptr == nullptr)
```

Comment:
Ultimately our style in C++ is determined by `clang-format -i [file]`, which says that it's if-space-openparen.

If you can run clang-format (which will fix some existing style violations in the file) that'd be great.  Otherwise, it'd be good to fix up some of these style problems so a later execution of it gets less noise.